src/System.Drawing.Primitives/src/System/Drawing/Color.cs

```diff
@@ -420,188 +420,119 @@ private long Value
+
+            return FromArgb(MakeArgb((byte)alpha, (byte)red, (byte)green, (byte)blue));
         }
 
         public static Color FromArgb(int alpha, Color baseColor)
         {
             CheckByte(alpha, nameof(alpha));
-            // unchecked - because we already checked that alpha is a byte in CheckByte above
-            return new Color(MakeArgb(unchecked((byte)alpha), baseColor.R, baseColor.G, baseColor.B), StateARGBValueValid, null, (KnownColor)0);
+
+            return FromArgb(unchecked((uint)alpha << ARGBAlphaShift | ((uint)baseColor.Value & ~ARGBAlphaMask)));
         }
 
-        public static Color FromArgb(int red, int green, int blue) => FromArgb(255, red, green, blue);
+        public static Color FromArgb(int red, int green, int blue) => FromArgb(byte.MaxValue, red, green, blue);
 
         public static Color FromKnownColor(KnownColor color) =>
             color <= 0 || color > KnownColor.MenuHighlight ? FromName(color.ToString()) : new Color(color);
 
         public static Color FromName(string name)
         {
             // try to get a known color first
-            Color color;
-            if (ColorTable.TryGetNamedColor(name, out color))
-            {
+            if (ColorTable.TryGetNamedColor(name, out Color color))
                 return color;
-            }
+
             // otherwise treat it as a named color
             return new Color(NotDefinedValue, StateNameValid, name, (KnownColor)0);
         }
 
         public float GetBrightness()
         {
-            float r = R / 255.0f;
-            float g = G / 255.0f;
-            float b = B / 255.0f;
+            uint value = (uint)Value;
+            int r = (int)((value >> ARGBRedShift  ) & 0xff);
+            int g = (int)((value >> ARGBGreenShift) & 0xff);
+            int b = (int)((value >> ARGBBlueShift ) & 0xff);
 
-            float max, min;
+            int min = Math.Min(Math.Min(r, g), b);
+            int max = Math.Max(Math.Max(r, g), b);
 
-            max = r; min = r;
-
-            if (g > max)
-            {
-                max = g;
-            }
-            else if (g < min)
-            {
-                min = g;
-            }
-
-            if (b > max)
-            {
-                max = b;
-            }
-            else if (b < min)
-            {
-                min = b;
-            }
-
-            return (max + min) / 2;
+            return (max + min) / (byte.MaxValue * 2f);
         }
 
 
         public float GetHue()
         {
-            if (R == G && G == B)
-                return 0; // 0 makes as good an UNDEFINED value as any
-
-            float r = R / 255.0f;
-            float g = G / 255.0f;
-            float b = B / 255.0f;
-
-            float max, min;
-            float delta;
-            float hue;
+            uint value = (uint)Value;
+            int r = (int)((value >> ARGBRedShift  ) & 0xff);
```

Comment:
Nit: the extra spaces before `)` we usually don't do that to inline consequent code lines.